### PR TITLE
Add require polyfill

### PR DESCRIPTION
**Pull request by yaeju1205**
make polyfill, tests

### CODE REVIEW

**Comment by yaeju1205** [comment]
Closes #15

**Comment by jiwonz** [comment]
I think it should be made following [Luau's latest require semantics](https://rfcs.luau.org/amended-require-resolution.html). including [the @self](https://rfcs.luau.org/abstract-module-paths-and-init-dot-luau.html) support

**Comment by yaeju1205** [comment]
Support `@self`

File: src/require.luau [hunk 1]
```diff
@@ -0,0 +1,7 @@
+local sub = string.sub
+local gsub = string.gsub
+
+return function(path)
+    gsub(path, "@self", "./")
```
Comment (jiwonz) [inline src/require.luau]: string is immutable in lua. this doesn't do anything

File: src/require.luau [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    return require(sub((debug :: any).getinfo(2, "S").source, 2)) :: any
```
Comment (jiwonz) [inline src/require.luau]: the argument `path` is never used. what even they do?

File: src/require.luau [hunk 3]
```diff
@@ -0,0 +1,7 @@
+local sub = string.sub
+local gsub = string.gsub
+
+return function(path)
```
Comment (jiwonz) [inline src/require.luau]: dalbit-polyfill is considered as Luau code but this is not typed

File: tests/polyfill.luau [hunk 4]
```diff
@@ -989,4 +989,14 @@ return function()
 			]]))
 		end)
 	end)
+
+	test.suite("require", function()
+		test.case("require function", function()
+			check.equal(utils.execute([[
+
+			return type(require "tests/utils".executeWithLuau)
```
Comment (jiwonz) [inline tests/polyfill.luau]: This shouldn't be successful if you're using the latest Luau cli binary or Lune. every path input for require should start in ./, ../, or, @ otherwise, they are all invalid.

Also checking/testing via `type` is kinda ambiguous because it can be `nil` or something error.

File: tests/polyfill.luau [hunk 5]
```diff
@@ -989,4 +989,14 @@ return function()
 			]]))
 		end)
 	end)
+
+	test.suite("require", function()
+		test.case("require function", function()
```
Comment (jiwonz) [inline tests/polyfill.luau:994]: Should add more cases for ./, ../, @

File: src/require.luau [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+    gsub(path, "@self", "./")
+    return require(sub((debug :: any).getinfo(2, "S").source, 2)) :: any
+end
```
Comment (jiwonz) [inline src/require.luau:37]: We need to add error messages for each invalid require paths

**Review by jiwonz — CHANGES_REQUESTED** [review]
Also, We should add an option to enable or disable the require polyfill so that dalbit users can choose it via a config file.
